### DISCUSSION TITLE
how do I fetch information like parameters and metrics from an experiment done before using python API

### DISCUSSION THREAD

**Discussion by Hikaru-13915** [mlflow/mlflow#7552] [Q&A] [answered]
Is there any way to fetch information like parameters and metrics from an experiment done before using python API?
I need something like `list=Get_experiment_by_ID(ex_ID).get_parameters()`, then extract accuracy and loss, all params recorded at the experiment.

**Comment by JHibbard** [comment, marked as answer]
Hi Hikaru-13915,

If I'm understanding your question correctly, you want to retrieve parameters and metrics from an experiment that you've run using the Python API? If so, I'd suggest using the [mlflow.search_runs](https://mlflow.org/docs/latest/python_api/mlflow.html#mlflow.search_runs) API. You can pass a list of experiment ids and you'll be returned all related Run objects which contain info on metrics and parameters under `Run.data.metrics` and `Run.data.params`. The RunData objects have a `.to_dictionary()` method which might also be useful.

Hope that helps!
Jim

**Reply by Hikaru-13915** [reply]
@JHibbard 
Thank you Mr.Jim!

Yes! using Python API.
I think what you have suggested is what I want.
I'm trying ot fetch info, as you have told me.
But I don't know how to do with `to_dictionary()` to fetch the matrics infomation.
Could you give me some example code with real values?

```
with mlflow.start_run():
    mlflow.doctor()
    print("=========================")
    print(mlflow.get_artifact_uri())

    entity=mlflow.search_runs(experiment_ids=[1,2])
    mlflow.entities.RunData(entity[1]).to_dictionary()
    mlflow.entities.Run().to_dictionary()
 ```
